src/cli/java/org/commcare/util/screen/SyncScreen.java

```diff
@@ -0,0 +1,140 @@
+
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.session.CommCareSession;
+import org.commcare.suite.model.Entry;
+import org.commcare.suite.model.PostRequest;
+import org.commcare.suite.model.RemoteRequestEntry;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Hashtable;
+
+import okhttp3.Credentials;
+import okhttp3.HttpUrl;
+import okhttp3.MultipartBody;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/**
+ * Screen to make a sync request to HQ after a case claim. Unlike other all other screens,
+ * SyncScreen does not take input - it simply processes the PostRequest object, makes the request,
+ * and displays the result (if necessary)
+ */
+public class SyncScreen extends Screen {
+
+    protected SessionWrapper sessionWrapper;
+    private String username;
+    private String password;
+    private PrintStream printStream;
+    private boolean syncSuccessful;
+
+    public SyncScreen(String username, String password, PrintStream printStream) {
+        super();
+        this.username = username;
+        this.password = password;
+        this.printStream = printStream;
+    }
+
+    @Override
+    public void init (SessionWrapper sessionWrapper) throws CommCareSessionException {
+        this.sessionWrapper = sessionWrapper;
+        parseMakeRequest();
+    }
+
+    private void parseMakeRequest() throws CommCareSessionException {
+        String command = sessionWrapper.getCommand();
+        Entry commandEntry = sessionWrapper.getPlatform().getEntry(command);
+
+        if (!(commandEntry instanceof RemoteRequestEntry)) {
+            // expected a sync entry; clear session and show vague 'session error' message to user
+            throw new CommCareSessionException("Initialized sync request while not on sync screen");
+        }
+
+        PostRequest syncPost = ((RemoteRequestEntry)commandEntry).getPostRequest();
+        try {
+            Response response = makeSyncRequest(syncPost);
+            if (!response.isSuccessful()) {
+                printStream.println(String.format("Sync request failed with response code %s and message %s", response.code(), response.body()));
+                printStream.println("Press 'enter' to retry.");
+                return;
+            }
+            syncSuccessful = true;
+            SessionUtils.restoreUserToSandbox(sessionWrapper.getSandbox(),
+                    sessionWrapper,
+                    sessionWrapper.getPlatform(),
+                    username,
+                    password);
+
+            printStream.println(String.format("Sync successful with response %s", response));
+            printStream.println("Press 'enter' to continue.");
+        } catch (IOException e) {
+            e.printStackTrace();
+            printStream.println(String.format("Sync failed with exception %s", e.getMessage()));
+            printStream.println("Press 'enter' to retry.");
```

Comment:
Hm, just realized this will leave the user in an infinite loop. Is it possible for us to take in an input which decides whether to retry? (It's possible that the Command infrastructure, ie - ":home" or whatever) allows this, already?

Sorry for all of the back and forth on this PR,I'm just thinking of mistakes I made in the CLI which ended up frustrating me a lot later and causing a lot of churn when I'd get deep into debugging something and the whole session would just die or become unsalvagable

Reply:
Sorry, I'm not seeing the infinite loop - where are you referring to? If I manually throw an IOException from `makeSyncRequest` then I am prompted to retry and can type `:home` in order to exit if need be.